code/datums/elements/weather_listener.dm

```diff
@@ -0,0 +1,45 @@
+///This element just handles creating and destroying an area sound manager that's hooked into weather stuff
+/datum/element/weather_listener
+	element_flags = ELEMENT_DETACH | ELEMENT_BESPOKE
```

Comment:
You're only unregistering a signal when you detach, you don't need to handle qdel explicitly as when they get destroyed the signals will be gone as well

Reply:
Ah you're right, thank you